logstash-core/lib/logstash/pipeline.rb

```diff
@@ -82,14 +82,13 @@ def initialize(config_str, settings = LogStash::SETTINGS, namespaced_metric = ni
       raise
     end
 
-    @input_queue = LogStash::Util::WrappedSynchronousQueue.new
+    queue = LogStash::Util::WrappedSynchronousQueue.new
+    @input_queue_client = queue.write_client
+    @filter_queue_client = queue.read_client
```

Comment:
so  I read that to @andrewvc 's question the answer is "yes", we could have per-input write clients. right?